### PR TITLE
refactor(v2): better mdx-loader

### PR DESCRIPTION
**Pull request by endiliey**
## Motivation

- Remove mdx-js/loader. we should write our own loader for customizability purpose
- Make docusaurus markdown-loader asynchronous. This is because expensive synchronous computations are a bad idea in a single-threaded environment like Node.js. Our markdown-loader is independent from each other, so let's make it asynchronous.
- Change syntax highlighting theme to atom dark
- Only import prism css file for pages that requires it

TODO: need to decide how user can choose the theme

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan
test

- Netlify

### CODE REVIEW

File: packages/docusaurus/lib/core/App.js [hunk 1]
```diff
@@ -13,9 +13,6 @@ import metadata from '@generated/metadata'; // eslint-disable-line
 import siteConfig from '@generated/docusaurus.config'; //eslint-disable-line
 import DocusaurusContext from '@docusaurus/context'; // eslint-disable-line
 
-// TODO: Allow choosing prismjs theme.
-import 'prismjs/themes/prism.css';
```
Comment (endiliey) [inline packages/docusaurus/lib/core/App.js]: We shouldn't import it globally

File: packages/docusaurus/lib/load/config.js [hunk 2]
```diff
@@ -98,8 +98,6 @@ function loadConfig(siteDir, deleteCache = true) {
   // e.g: if they want to include some.field so they can access it later from `props.siteConfig`.
   const {customFields = []} = config;
 
-  // TODO: Check that plugins mentioned exist.
```
Comment (endiliey) [inline packages/docusaurus/lib/load/config.js]: Already done https://github.com/facebook/Docusaurus/blob/d51f2c9f92d25771dbc272b8f9f8bac7a5a06937/packages/docusaurus/lib/load/plugins.js#L22

File: packages/docusaurus/lib/webpack/loaders/markdown/index.js [hunk 3]
```diff
@@ -6,12 +6,16 @@
...
 
 const fm = require('front-matter');
+const mdx = require('@mdx-js/mdx');
 const {getOptions} = require('loader-utils');
 const path = require('path');
 const {resolve} = require('url');
 
-module.exports = function(fileString) {
-  const options = getOptions(this);
+module.exports = async function(fileString) {
```
Comment (endiliey) [inline packages/docusaurus/lib/webpack/loaders/markdown/index.js]: Make this asynchronous loader

Comment (yangshun) [reply packages/docusaurus/lib/webpack/loaders/markdown/index.js]: Interesting, I didn't know this was possible.

**Pull request merged**